subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaExecAction.java

```diff
@@ -44,23 +46,29 @@
  */
 public class DefaultJavaExecAction implements JavaExecAction {
 
-    private final JavaExecSpec javaExecSpec;
+    private final DefaultJavaExecSpec javaExecSpec;
     private final JavaExecHandleBuilder javaExecHandleBuilder;
+    @Nullable
+    private final JavaModuleDetector javaModuleDetector;
 
     @Inject
-    public DefaultJavaExecAction(JavaExecSpec javaExecSpec, JavaExecHandleBuilder javaExecHandleBuilder) {
+    public DefaultJavaExecAction(
+        DefaultJavaExecSpec javaExecSpec,
+        JavaExecHandleBuilder javaExecHandleBuilder
+    ) {
         this.javaExecSpec = javaExecSpec;
         this.javaExecHandleBuilder = javaExecHandleBuilder;
+        this.javaModuleDetector = javaExecHandleBuilder.getJavaModuleDetector();
```

Comment:
❓ What is the point of storing this reference here? IIUC, `javaModuleDetector` is an immutable field of `JavaExecHandlerBuilder`, no?

Reply:
Good point, I can remove it, done

Reply:
Done